oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -405,26 +406,32 @@ sshmem_mkey_t *mca_spml_ucx_register(void* addr,
         return NULL;
     }
 
-    seg = memheap_find_segnum(addr);
+    segno   = memheap_find_segnum(addr);
```

Comment:
It is always valid because the register function is only called for memory segments that were already allocated and belong to the memheap.